JavaScript/JavaScriptSDK/Telemetry/Timings.ts

```diff
@@ -0,0 +1,51 @@
+﻿/// <reference path="../Util.ts"/>
+
+module Microsoft.ApplicationInsights.Telemetry {
+    "use strict";
+
+    export class Timings {
+
+        public duration: number;
+        public perfTotal: string;
+        public networkConnect: string;
+        public sentRequest: string;
+        public receivedResponse: string;
+        public domProcessing: string;
+
+        constructor(durationMs: number, networkMs: number, sendMs: number, receiveMs: number, domProcessingMs: number) {
+            this.duration = 0;
+            var total = Timings.getDuration(durationMs);
+            var network = Timings.getDuration(networkMs);
+            var request = Timings.getDuration(sendMs);
+            var response = Timings.getDuration(receiveMs);
+            var dom = Timings.getDuration(domProcessingMs);
+            
+            if (total < Math.floor(network) + Math.floor(request) + Math.floor(response) + Math.floor(dom)) {
```

Comment:
Looks like Timings acts as a container which you fill out and pass to trackPageView. So it shouldn't do any checks here as they are done in PageViewPerformance.